File: src/main/kotlin/com/lcaohoanq/sp/domains/notifications/FcmTokenService.java
package com.lcaohoanq.sp.domains.notifications;


import com.lcaohoanq.sp.domains.notifications.FcmTokenPort.CreateUserDeviceTokenReq;
import com.lcaohoanq.sp.domains.notifications.FcmTokenPort.UpdateUserDeviceTokenReq;
import com.lcaohoanq.sp.domains.notifications.FcmTokenPort.UserDeviceTokenDto;
import java.util.List;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface FcmTokenService {
    Page<UserDeviceTokenDto> getAll(Pageable pageable);
    UserDeviceTokenDto getById(Long id);
    UserDeviceTokenDto create(CreateUserDeviceTokenReq req);
    UserDeviceTokenDto update(Long id, UpdateUserDeviceTokenReq req);
    UserDeviceTokenDto createOrUpdate(CreateUserDeviceTokenReq req);
    List<String> getFcmTokensByEmail(String email);
}

File: src/main/kotlin/com/lcaohoanq/sp/domains/notifications/FcmTokenService.kt
package com.lcaohoanq.sp.domains.notifications

import com.lcaohoanq.sp.domains.notifications.FcmTokenPort.*
import com.lcaohoanq.sp.entities.UserDeviceToken
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable


interface FcmTokenService {
    fun getAll(pageable: Pageable): Page<UserDeviceToken>
    fun getById(id: Long): UserDeviceToken
    fun create(req: CreateUserDeviceTokenReq): UserDeviceToken
    fun update(id: Long, req: UpdateUserDeviceTokenReq): UserDeviceToken
    fun createOrUpdate(req: CreateUserDeviceTokenReq): UserDeviceToken
    fun getFcmTokensByEmail(email: String): MutableList<String>
}
